BlueprintUI/Sources/ViewDescription/LifecycleCallback.swift

```diff
@@ -0,0 +1,46 @@
+import Foundation
+
+/// This is the type used by all lifecycle callback hooks.
+///
+/// - Tag: ElementLifecycle
+///
+/// # The Element Lifecycle
+///
+/// View-backed elements have a lifecycle with 3 states:
+///
+/// 1. **Unmounted**
+///
+///    This is the initial state of an element, before a render pass.
+///
+/// 2. **Mounted**
```

Comment:
Maybe just me; but I don't love the mounted/unmounted terminology; maybe just `new/loaded/visible` or something?

Reply:
> Maybe just me; but I don't love the mounted/unmounted terminology; maybe just `new/loaded/visible` or something?

Borrowed from React. We're going to have to define exactly what it means either way, I think.

Reply:
I guess in general I don't like things like "mount" or "render" since they're not common terms in iOS land at all – maybe some naming we could borrow from SwiftUI or something?

Reply:
(Maybe `create` and `teardown` or similar?)

Reply:
> I guess in general I don't like things like "mount" or "render" since they're not common terms in iOS land at all – maybe some naming we could borrow from SwiftUI or something?

SwiftUI doesn't have an analogous concept to mounting. Not sure if that's because of differences in the layout system, or if it's just something that's not exposed.

I'm happy to use existing terminology when it exists, but I don't think it's helpful to repurpose familiar terms for unfamiliar concepts.

If other folks find the mount/unmount idea confusing, perhaps we should not expose these events, or the mounted state, at all? Personally, I think that since the appear/disappear events do not directly correspond with views being created/added/removed/released, that middle state of "mounted" is useful for representing the difference, but maybe it's just overcomplicating the API?